mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -729,14 +735,18 @@ pub async fn insert_new_swap_to_db(
 }
 
 #[cfg(not(target_arch = "wasm32"))]
-fn add_swap_to_db_index(ctx: &MmArc, swap: &SavedSwap) {
-    crate::mm2::database::stats_swaps::add_swap_to_index(&ctx.sqlite_connection(), swap)
+fn add_swap_to_db_index(ctx: &MmArc, swap: &SavedSwap, coins_price: Option<crate::mm2::lp_price::CEXRates>) {
+    crate::mm2::database::stats_swaps::add_swap_to_index(&ctx.sqlite_connection(), swap, coins_price);
 }
 
 #[cfg(not(target_arch = "wasm32"))]
-async fn save_stats_swap(ctx: &MmArc, swap: &SavedSwap) -> Result<(), String> {
+async fn save_stats_swap(
+    ctx: &MmArc,
+    swap: &SavedSwap,
+    coins_price: Option<crate::mm2::lp_price::CEXRates>,
+) -> Result<(), String> {
     try_s!(swap.save_to_stats_db(ctx).await);
-    add_swap_to_db_index(ctx, swap);
+    add_swap_to_db_index(ctx, swap, coins_price);
```

Comment:
I have a few suggestions how to improve the code:
1) Please add `SavedSwap::fetch_and_set_usd_prices`, `MakerSavedSwap::fetch_and_set_usd_prices` and `TakerSavedSwap::fetch_and_set_usd_prices` (or choose another method name):
```rust
    pub async fn fetch_and_set_usd_prices(&mut self) {
        if let Some(rates) = fetch_swap_coins_price(self.maker_coin.clone(), self.taker_coin.clone()).await {
            self.maker_coin_usd_price = Some(rates.base);
            self.taker_coin_usd_price = Some(rates.rel);
        }
    }
```
Such methods can be used on [save_my_maker_swap_event](https://github.com/KomodoPlatform/atomicDEX-API/blob/store-fiat-price-on-the-moment-of-trade/mm2src/mm2_main/src/lp_swap/maker_swap.rs#L98-L103) and [save_my_taker_swap_event](https://github.com/KomodoPlatform/atomicDEX-API/blob/store-fiat-price-on-the-moment-of-trade/mm2src/mm2_main/src/lp_swap/taker_swap.rs#L96-L101)

2) I missed that we also save the `swap` status to filesystem/IndexedDb on
https://github.com/KomodoPlatform/atomicDEX-API/blob/7e413314612ce036e79449ff3336fec76bb85bba/mm2src/mm2_main/src/lp_swap.rs#L748
So you made the right decision to request the prices, even if the `swap` is already in the DB index.
Since `SavedSwap::fetch_and_set_usd_prices` sets the coin prices (mutates itself), we can avoid taking the `coins_price: Option<crate::mm2::lp_price::CEXRates>` argument at `lp_swap:: save_stats_swap`, `lp_swap::add_swap_to_db_index`, `stats_swaps::add_swap_to_index` etc functions.

Consuming up, it's just needed to call `SavedSwap::fetch_and_set_usd_prices` before `lp_swap::save_stats_swap`:

```rust
// lp_swap.rs
// fn process_message
            #[cfg(not(target_arch = "wasm32"))]
            match json::from_slice::<SwapStatus>(msg) {
                Ok(mut status) => {
                    status.data.fetch_set_usd_prices().await;
                    if let Err(e) = save_stats_swap(&ctx, &status.data).await {
                        error!("Error saving the swap {} status: {}", status.data.uuid(), e);
                    };
                },
                Err(swap_status_err) => {
                    error!("Couldn't deserialize 'SwapMsg': {:?}", swap_msg_err);
                    error!("Couldn't deserialize 'SwapStatus': {:?}", swap_status_err);
                },
            };
```

**Important note**
As I can see, you fetch USD prices on the [broadcast_my_swap_status](https://github.com/KomodoPlatform/atomicDEX-API/blob/store-fiat-price-on-the-moment-of-trade/mm2src/mm2_main/src/lp_swap.rs#L879-L883) function call.
Please note that `broadcast_my_swap_status` is called within `run_maker_swap` right after [save_my_maker_swap_event](https://github.com/KomodoPlatform/atomicDEX-API/blob/store-fiat-price-on-the-moment-of-trade/mm2src/mm2_main/src/lp_swap/maker_swap.rs#L1769-L1789) (same for Taker swap).
So the prices should be set already, and there is no need to request them again.